### PR TITLE
Scala cmdl compiler

### PR DESCRIPTION
**Pull request by codepitbull**
Resolves https://github.com/vert-x3/vertx-lang-scala/issues/11

### CODE REVIEW

File: vertx-lang-scala-on-the-fly/src/main/scala/io/vertx/lang/scala/onthefly/CompMain.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package io.vertx.lang.scala.onthefly
+
+/**
+  * Created by jochen on 17.03.17.
+  */
+object CompMain {
+  def main(args: Array[String]): Unit = {
+    val compiler = new OnTheFlyCompiler(None)
+    val clazz = compiler
+      .tryToCompileClass("/Users/jochen/Development/8_vertx/vertx-lang-scala/vertx-lang-scala-on-the-fly/src/test/resources/SourceClass.scala")
```
Comment (vietj) [inline vertx-lang-scala-on-the-fly/src/main/scala/io/vertx/lang/scala/onthefly/CompMain.scala]: isn't that a problem ?

Comment (codepitbull) [inline vertx-lang-scala-on-the-fly/src/main/scala/io/vertx/lang/scala/onthefly/CompMain.scala]: Holy cow, overlooked that one.
PR ftw :)

Comment (codepitbull) [inline vertx-lang-scala-on-the-fly/src/main/scala/io/vertx/lang/scala/onthefly/CompMain.scala]: Luckily that was a class I only used for my first test. I deleted it.

**Comment by tsegismont** [comment]
Looks good to me. Only one minor comment, when reading the class name:

```scala
className.split(System.getProperty("file.separator")).last.replace(".scala","")
```
Wouldn't it be safer to do this with Java's `Path` or a Scala equivalent, in case the path mixes e.g Win and Unix paths separators?

Also, it might not be a problem in Scala but what happens if the verticle is in a package? The test case is for a verticle placed in the root of the classpath.

**Comment by codepitbull** [comment]
Thanks for pointing that out. I will take a closer look at how the other langs handle that case.
I have the feeling I missed a few cases.

**Comment by codepitbull** [comment]
Thanks for the recommendation. I switched to using Paths and added the capability to build Scala-Files from the classpath.

**Comment by tsegismont** [comment]
:+1:

**Pull request merged**